Add support for buttons in Alerts

BoredManCodes opened the pull request:
This PR adds the ability for you to add buttons to alerts, buttons can be URL buttons, or styled buttons that run a command or an array of commands.
Buttons can now be used in alerts like so:
```
- Trigger: net.boredman.prismPlugin.events.NewPlayerQuitEvent
  Channel: modlog
  Embed:
    Color: "#869600"
    ThumbnailUrl: "https://heads.discordsrv.com/head.png?name=${#event.playerName}&overlay"
    Title:
      Text: "New Member Quit"
    Description: "Please investigate their actions."
    Fields:
      - "Minecraft Name:;${#event.playerName};true"
      - "Played For:;${#event.playtimeHuman};true"
  Buttons:
    - Text: "View NameMC"
      Url: "https://namemc.com/profile/${#event.playerName}"
    - Text: "ROLLBACK"
      Style: "danger"
      Command: "co rollback radius:#GLOBAL user:${#event.playerName} time:1h"
      RunAs: "console"
    - Text: "BAN"
      Style: "danger"
      Commands: 
        - "ban ${#event.playerName} One click ban as new member"
        - "discord bcast #897765157940396052 Please reply to this message with the ban reason"
      RunAs: "console"
```
<img width="625" height="214" alt="image" src="https://github.com/user-attachments/assets/a1c0b8a6-73bd-41ca-bd48-c4df4efd5ffb" />

BoredManCodes committed:
Add support for link buttons

Buttons can be used in alerts like so:
```
- Trigger: github.scarsz.discordsrv.api.events.AccountLinkedEvent
  Channel: global
  Embed:
    Enabled: true
    Color: "#ffffff"
    Author:
      ImageUrl: "{embedavatarurl}"
      Name: "User '${#event.getPlayer().getName()}' linked their account with the discord user '${#event.getUser().getAsTag()}'"
    Timestamp: true
  Buttons:
    - Text: "View Player UUID"
      Url: "https://mcuuid.net/?q=${#event.getPlayer().getUniqueId()}"
      Style: LINK
    - Text: "Server Website"
      Url: "https://example.com" # replace with your site
      Style: LINK
```

BoredManCodes committed:
Added command buttons alongside link buttons

Example usage:
```yml
- Trigger: org.bukkit.event.player.PlayerJoinEvent
  Channel: global
  Embed:
    Color: "#00FF00"
    Author:
      ImageUrl: "{embedavatarurl}"
      Name: "{displayname} joined"
  Buttons:
      # Command button (non-link), style can be primary/secondary/success/danger
      - Text: "Kick Player"
        Style: "danger"
        Command: "kick {name} Kicked by staff via Discord"
        RunAs: "console"  # optional; console is default, included here for clarity

      # Link button: style is automatically LINK and other styles are ignored
      - Text: "View UUID"
        Url: "https://mcuuid.net/?q={name}"
```

Scarsz commented on src/main/java/github/scarsz/discordsrv/modules/alerts/AlertListener.java:
```diff
@@ -286,6 +312,96 @@ public void unregister() {
+                        final String finalPlayerName = playerName;
+                        SchedulerUtil.runTask(DiscordSRV.getPlugin(), () -> {
+                            try {
+                                Player p = finalPlayerName != null ? Bukkit.getPlayerExact(finalPlayerName) : null;
```
I don't think it's useful to be able to specify player names directly for the command targets. Nobody besides the event player themselves are guaranteed to be online, and even then there's possibilities the player might not be. I'd suggest just changing this to a `player` or `console` switch with `player` automatically invoking an event's #getPlayer() method if it exists. The player lookup can stay as a fallback for events that offer the player names in different methods though

BoredManCodes committed:
Add ability for an array of commands to be executed

BoredManCodes committed:
Address review feedback on alert button changes

- Restore HandlerList.allLists proxy so dynamic handler registration still
  works; on Java versions where modifiersField reflection fails, fall back
  to a one-shot scan plus a PluginEnableEvent re-scan so plugins loaded
  later still get the alert listener attached.
- Claim button actions atomically via map.remove() to prevent double
  execution from rapid clicks, and reply with an expiry message when the
  mapping is gone (expired or lost across restart) instead of silently
  acknowledging.
- Prefix button custom IDs with "dsrv-alert:" so foreign ButtonClickEvents
  fall through to the normal alert path.
- Store target as UUID and add per-button FallbackToConsole flag (default
  false) so RunAs: player no longer silently elevates to console when the
  player is offline.
- Extract Discord component caps into named constants, warn explicitly
  on missing required Text field, narrow Throwable catches to Exception,
  fix import order.
